generator/src/generator.rs

```diff
@@ -0,0 +1,767 @@
+                data: AdditionalData::new(vec![])?,
+            };
+
+            let payload = UpdateContractPayload {
+                amount:       Amount::from_micro_ccd(1),
+                address:      contract_address,
+                receive_name: OwnedReceiveName::new("cis2_wCCD.wrap".into())?,
+                message:      OwnedParameter::from_serial(&params)?,
+            };
+
+            let wrap_tx = args.make_update_contract_transaction(
+                payload,
+                Energy::from(3500),
+                &mut nonce.nonce,
+            )?;
+            let block_item = BlockItem::AccountTransaction(wrap_tx);
+            let transaction_hash = args.client.send_block_item(&block_item).await?;
+            println!(
+                "{}: Transferred 1 wCCD to {recv} (hash = {transaction_hash}).",
+                chrono::Utc::now(),
+            );
+        }
+
+        Ok(Self {
+            args,
+            contract_address,
+            nonce: nonce.nonce,
+            count: 0,
+        })
+    }
+}
+
+impl Generate for WccdGenerator {
+    fn generate(&mut self) -> anyhow::Result<AccountTransaction<EncodedPayload>> {
+        // We modulate between wrapping, transferring, and unwrapping. All wCCD are
+        // minted for and transferred to our own account, which is fine for testing,
+        // since there is no special logic for this in the contract.
+        let payload = match self.count % 3 {
+            // Wrap
+            0 => {
+                let params = WrapParams {
+                    to:   Receiver::Account(self.args.keys.address),
+                    data: AdditionalData::new(vec![])?,
+                };
+
+                UpdateContractPayload {
+                    amount:       Amount::from_micro_ccd(1),
+                    address:      self.contract_address,
+                    receive_name: OwnedReceiveName::new("cis2_wCCD.wrap".into())?,
+                    message:      OwnedParameter::from_serial(&params)?,
+                }
+            }
+            // Transfer
+            1 => {
+                let params = TransferParams::new(
+                    [Transfer {
+                        // The token id of wCCD is 0.
+                        token_id: TokenId::new(vec![])?,
+                        amount:   TokenAmount::from(1u32),
+                        from:     Address::Account(self.args.keys.address),
+                        to:       Receiver::Account(self.args.keys.address),
+                        data:     AdditionalData::new(vec![])?,
+                    }]
+                    .to_vec(),
+                )?;
+
+                UpdateContractPayload {
+                    amount:       Amount::zero(),
+                    address:      self.contract_address,
+                    receive_name: OwnedReceiveName::new("cis2_wCCD.transfer".into())?,
+                    message:      OwnedParameter::from_serial(&params)?,
+                }
```

Comment:
Please use cis2Client transfer here instead.

Reply:
See my other comment about this.

Reply:
There are now `make_transfer*` helpers in Cis2Contract (and similar ones in CIS4Contrac) that just construct the transaction to send.